gis/geo_sir/model.py

```diff
@@ -102,9 +94,11 @@ def reset_counts(self):
 
     def step(self):
         """Run one step of the model."""
-        self.steps += 1
         self.reset_counts()
-        self.schedule.step()
+
+        # Activate PersonAgents first, in random order, then activate NeighbourhoodAgents.
+        self.agents_by_type[PersonAgent].shuffle_do("step")
+        self.agents_by_type[NeighbourhoodAgent].do("step")
```

Comment:
Why does the ordering not matter here? Anyone reading the code would have the same question.

Reply:
NeighbourhoodAgents work independently from each other. I will add a comment.

(note it’s already way more transparent then with the scheduler)